### PR TITLE
[1580] use callbacks to add or remote sites to course

### PR DESCRIPTION
**Pull request by misaka**
- this replaces the functionality of add_site! and remove_site!

### Context

We should be able to add or remove sites to a course without having to resort to custom methods like `add_site!` or `remove_site!`.

### Changes proposed in this pull request

Replace the functionality in those methods with callbacks in `SiteStatus`.

### Guidance to review

This code was done as a proof of concept and, on account of it being pretty late and me not being able to see straight, could probably use a little tidying up. It also doesn't remove the mentioned methods, that might be slightly more work.

This change could probably happen independent of `1580-edit-locations-validations`, if we think it's a good idea.

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: app/models/site_status.rb [hunk 1]
```diff
@@ -19,6 +20,21 @@ class SiteStatus < ApplicationRecord
 
   after_initialize :set_defaults
   before_validation :set_vac_status
+  before_create :set_new_status
+
+  def set_new_status
+    if !course.site_statuses.empty? && !course.site_statuses.all?(&:status_new_status?)
+      self.status = 'R'
```
Comment (benilovj) [inline app/models/site_status.rb]:
```suggestion
      self.status = : running
```

Comment (defong) [reply app/models/site_status.rb]: Just wondering that this change will affect the factory...

**Comment by misaka** [comment]
I'm cool with this being separate. I'll prep it up now, and probably give it a new card.

File: app/models/site_status.rb (hunk 1 above)
Comment (misaka) [reply app/models/site_status.rb]: @defong I'll have a look.

**Comment by misaka** [comment]
Will re-cut this PR from another branch

**Pull request closed without merging**